instrumentation/opentelemetry_bandit/lib/opentelemetry_bandit.ex

```diff
@@ -1,178 +1,515 @@
-      kind: :error,
-      status: :error
-    })
-    |> set_span_status(meta, inspect(meta.stacktrace))
-    |> OpenTelemetry.Span.end_span()
+  defp set_network_protocol_attrs(attrs, conn) do
+    case extract_network_protocol(conn.adapter) do
+      {:error, _reason} ->
+        attrs
 
-    OpenTelemetry.Ctx.clear()
+      {:http, version} ->
+        Map.put(attrs, NetworkAttributes.network_protocol_version(), version)
+
+      {protocol, version} ->
+        Map.merge(attrs, %{
+          NetworkAttributes.network_protocol_name() => protocol,
+          NetworkAttributes.network_protocol_version() => version
+        })
+    end
   end
 
-  def handle_websocket_stop(_event, measurements, meta, _config) do
-    duration = measurements.duration
-    end_time = :opentelemetry.timestamp()
-    start_time = end_time - duration
+  defp set_query_string_attr(attrs, %{query_string: ""}), do: attrs
 
-    attributes = %{
-      :"websocket.recv.binary.frame.bytes" => Map.get(measurements, :send_binary_frame_bytes, 0),
-      :"websocket.send.binary.frame.bytes" => Map.get(measurements, :recv_binary_frame_bytes, 0),
-      Trace.net_transport() => :websocket
-    }
+  defp set_query_string_attr(attrs, %{query_string: query}),
+    do: Map.put(attrs, URLAttributes.url_query(), query)
 
-    span_kind = if Map.has_key?(meta, :error), do: :error, else: :server
+  # server attributes are only set from headers
+  # https://opentelemetry.io/docs/specs/semconv/http/http-spans/#setting-serveraddress-and-serverport-attributes
+  defp set_server_address_attrs(attrs, conn, config) do
+    case extract_server_address(conn, config) do
+      %{address: :undefined} ->
+        attrs
 
-    OpenTelemetry.Tracer.start_span("Websocket", %{
-      attributes: attributes,
-      start_time: start_time,
-      end_time: end_time,
-      kind: span_kind
-    })
-    |> set_span_status(meta, Map.get(meta, :error, ""))
-    |> OpenTelemetry.Span.end_span()
+      %{address: address, port: :undefined} ->
+        Map.put(attrs, ServerAttributes.server_address(), address)
 
-    OpenTelemetry.Ctx.clear()
+      %{address: address, port: port} ->
+        Map.merge(
+          attrs,
+          %{
+            ServerAttributes.server_address() => address,
+            ServerAttributes.server_port() => port
+          }
+        )
+    end
   end
 
-  defp set_span_status(span, meta, message) do
-    status = if Map.has_key?(meta, :error) || message != "", do: :error, else: :ok
+  defp extract_network_protocol(
+         {_adapter_name, %{transport: %{__struct__: Bandit.HTTP1.Socket} = transport}}
+       ) do
+    case transport.version do
+      :"HTTP/1.0" -> {:http, :"1.0"}
+      :"HTTP/1.1" -> {:http, :"1.1"}
+      :"HTTP/2.0" -> {:http, :"2"}
+      :"HTTP/2" -> {:http, :"2"}
+      :SPDY -> {:spdy, :"2"}
+      :QUIC -> {:quic, :"3"}
+      nil -> {:error, "Invalid protocol"}
```

Comment:
I imagine this code would never be a problem in practice, but it looks like `Bandit.HTTP1.Socket` [would never actually be used for anything but HTTP 1 / 1.1](https://github.com/mtrudel/bandit/blob/main/lib/bandit/http1/socket.ex#L36). Maybe this could be simpler?

I didn't see anything from a skim of Bandit  about SPDY or QUIC support.

So maybe we just have a catch-all here so that it returns an error for anything weird / unexpected, including `nil`? 🤷 
```suggestion
      _ -> {:error, "Invalid protocol"}
```